mods/ts/rules/world.yaml

```diff
@@ -198,47 +198,39 @@
 		Name: Nod
 		InternalName: nod
 		Description: Brotherhood of Nod\nThe Brotherhood is a religious cult centered around their leader Kane\nand the alien substance Tiberium. They utilize stealth technology\nand guerilla tactics to defeat those who oppose them.
-	ResourceType@Tiberium:
-		Type: Tiberium
-		Name: Tiberium
-		ResourceType: 1
-		Palette: greentiberium
-		Sequences: tib01, tib02, tib03, tib04, tib05, tib06, tib07, tib08, tib09, tib10, tib11, tib12
-		MaxDensity: 12
-		ValuePerUnit: 25
-		AllowedTerrainTypes: Clear, Rough, DirtRoad
-		AllowUnderActors: true
-		TerrainType: Tiberium
-	ResourceType@BlueTiberium:
-		Type: BlueTiberium
-		Name: Tiberium
-		ResourceType: 2
-		Palette: bluetiberium
-		Sequences: tib01, tib02, tib03, tib04, tib05, tib06, tib07, tib08, tib09, tib10, tib11, tib12
-		MaxDensity: 12
-		ValuePerUnit: 40
-		AllowedTerrainTypes: Clear, Rough, DirtRoad
-		AllowUnderActors: true
-		TerrainType: BlueTiberium
-	ResourceType@Veins:
-		Type: Veins
-		Name: Veins
-		ResourceType: 3
-		Palette: player
-		Sequences: veins
-		MaxDensity: 1
-		ValuePerUnit: 0
-		AllowedTerrainTypes: Clear, Rough, DirtRoad
-		AllowUnderActors: true
-		TerrainType: Veins
 	TerrainGeometryOverlay:
 	DebugVisualizations:
 	ExitsDebugOverlayManager:
 	CliffBackImpassabilityLayer:
 	SubterraneanActorLayer:
 	JumpjetActorLayer:
-	ResourceRenderer:
-		RenderTypes: Tiberium, BlueTiberium, Veins
+	TSTiberiumRenderer:
+		ResourceTypes:
+			Tiberium:
+				Sequences: tib01, tib02, tib03, tib04, tib05, tib06, tib07, tib08, tib09, tib10, tib11, tib12
+				Palette: greentiberium
+				Name: Tiberium
+			BlueTiberium:
+				Sequences: tib01, tib02, tib03, tib04, tib05, tib06, tib07, tib08, tib09, tib10, tib11, tib12
+				Palette: bluetiberium
+				Name: Tiberium
+		Ramp1Sequences:
```

Comment:
For this setup to be consistent with itself, these should really be defined on each ResourceType. Probably overkill for such a non-generic trait though.

Reply:
`ResourceTypes` is inherited from `ResourceLayer` so can't be overridden without introducing massive code duplication. Unless i'm missing something obvious?

Reply:
`TSTiberiumRenderer(Info)` inherits the property `ResourceTypes` (of type `ResourceRendererInfo.ResourceInfo`) from `ResourceRenderer(Info)`. Not to be confused with the `ResourceTypes` property of `ResourceLayerInfo.ResourceInfo` on `ResourceLayerInfo`. 😄 
Apart from that, it should be doable to override said property and also add a `public class TiberiumInfo : ResourceInfo` inside `TSTiberiumRendererInfo`, but like I said - very likely not worth in this particular case.

Reply:
It would be nice if it were this simple, but `ResourceTypes` is a field that it magically populated using a static method, so there's no room for virtual/overrides. We could use `new` to hide the base implementation, but then we would be parsing the resource types twice and different traits would be using different versions which I think is asking for trouble.

Reply:
The one thing we could do is to split out an abstract `ResourceRendererBase` trait / traitinfo that both inherit individually. This adds a bit more duplication and complexity, but perhaps justified here?

Reply:
It would definitely be an improvement to do so. I don't know how useful would the added flexibility be in practice though (for community mods ofc, not for our default ones). So up to you.

Reply:
I can't find a way to implement this that wouldn't be a total nightmare, so will skip it.